api/src/services/import-services/observation/utils/observation-sampling-row-validator.ts

```diff
@@ -47,6 +52,22 @@ export function getObservationSamplingInformationRowValidator(
     // Determine if the worksheet contains any sampling information
     const worksheetHasSamplingInformation = Boolean(worksheetSiteName || worksheetTechniqueName || worksheetPeriod);
 
+    // VALID: Sample period id is provided and exists in the list of sampling periods
+    if (samplePeriodId) {
+      if (findMatchingPeriodWithSamplePeriodId(samplingPeriods, samplePeriodId)) {
+        updateCSVRowState(params.row, { sample_period_id: samplePeriodId });
+
+        return [];
+      }
+
+      // Note: Normally we don't want to throw errors in validators, but in this case, it's a critical error
+      // that should be caught early in the import process
```

Comment:
nit: I would maybe expand on why its a critical error.

Something along the lines of like: 'because the sample period id is provided directly by the caller, we expect it to be valid"

Reply:
Updated